set.mm

```diff
@@ -704453,17 +704504,110 @@ is a wff not satisfied by a (unique) value ` x ` (there is no set, and
 
 $(
 -.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
-  Unordered and ordered pairs - extension for unordered pairs
+  Functions - extension
 -.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
 $)
 
-  $( An element of a proper unordered pair is the first element iff it is not
-     the second element.  (Contributed by AV, 18-Jun-2020.) $)
-  elprneb $p |- ( ( A e. { B , C } /\ B =/= C ) -> ( A = B <-> A =/= C ) ) $=
-    ( cpr wcel wne wceq wb wo wi elpri neeq1 eqcoms pm5.1 ex sylbid neeq2 wa wn
-    nesym sylan2b necon2abid sylbird jaoi syl imp ) ABCDEZBCFZABGZACFZHZUGUIACG
-    ZIUHUKJZABCKUIUMULUIUHUJUKUHUJHBABACLMUIUJUKUIUJNOPULUHBAFZUKACBQULUNUKULUN
-    RUIACUNULUISZULUOHBATULUONUAUBOUCUDUEUF $.
+  $( If a function's value at an argument is the universal class (which can
+     never be the case because of ~ fvex ), the function's value at this
+     argument is any set (especially the empty set).  In short "If a function's
+     value is a proper class, it is a set", which sounds strange/contradictory,
+     but which is a consequence of that a contradiction implies anything (see
+     ~ pm2.21i ).  (Contributed by Alexander van der Vekens, 26-May-2017.) $)
+  fveqvfvv $p |- ( ( F ` A ) = _V -> ( F ` A ) = B ) $=
+    ( cfv wceq cvv wcel wi fvex eleq1a ax-mp vprc pm2.21i syl eqcoms ) ACDZBEZF
+    PFPEZFFGZQPFGRSHACIPFFJKSQLMNO $.
+
+  $( Composition of two functions, generalization of ~ funco .  (Contributed by
+     Alexander van der Vekens, 25-Jul-2017.) $)
+  funresfunco $p |- ( ( Fun ( F |` ran G ) /\ Fun G ) -> Fun ( F o. G ) ) $=
+    ( crn cres wfun ccom funco wss wceq ssid cores ax-mp eqcomi funeqi sylibr
+    wa ) ABCZDZEBEPRBFZEABFZERBGTSSTQQHSTIQJABQKLMNO $.
+
+  $( Composition of two functions, similar to ~ fnco .  (Contributed by
+     Alexander van der Vekens, 25-Jul-2017.) $)
+  fnresfnco $p |- ( ( ( F |` ran G ) Fn ran G /\ G Fn B )
+                    -> ( F o. G ) Fn B ) $=
```

Comment:
Can be strengthened to:
```
fnresfnco $p |- ( ( Fun ( F |` ran G ) /\ G Fn B ) -> ( F o. G ) Fn B ) $=
```
from which the above is directly proved.

Reply:
Is this really true? Let `(dom F i^i ran G ) = (/) `, then `` ( F |` ran G ) = (/) ``, so that ``Fun ( F |` ran G )`` is true (~fun0). But 
` ( F o. G ) = (/) `in this case (~coeq0) and therefore ` B = (/) ` (~fn0). Therefore, this theorem holds only in the (degenerate) case `B = (/) ` ( G Fn B => G Fn (/) => G = (/) => ( F o. G ) = ( F o. (/) ) = (/) => ( F o. G ) Fn B because of (/) Fn (/) ) or if `ran G C_ dom F `).

Reply:
Ah, ok. Forgot the definition of Fun.